app/src/screens/Dialer/__tests__/Dialer.test.tsx

```diff
@@ -25,50 +26,195 @@ describe('<Dialer />', () => {
+        expect(callButton.props.accessibilityState.disabled).toBe(true);
+      });
+
+      it('should display an enabled client button', () => {
+        const clientButton = screen.getByTestId('client_button');
+        expect(clientButton).toBeOnTheScreen();
+        expect(clientButton.props.accessibilityState.disabled).toBe(false);
+        expect(clientButton.findByType(Text).props.children).toBe('Client');
+      });
+
+      it('should hide the backspace button', () => {
+        expect(screen.queryByTestId('back_space_button')).not.toBeOnTheScreen();
+      });
+    });
+
+    describe('press the first dialpad button', () => {
+      beforeEach(() => {
+        fireEvent.press(screen.getByText('4'));
+      });
+
+      it('should display "+" followed by the digit of the dialpad button', () => {
+        const display = screen.getByTestId('formatted_number');
+        expect(display.props.children).toBe('+4');
+      });
+
+      it('should enable the call button', () => {
+        const callButton = screen.getByTestId('call_button');
+        expect(callButton.props.accessibilityState.disabled).toBe(false);
+      });
+
+      it('should show the backspace button', () => {
+        expect(screen.getByTestId('back_space_button')).toBeOnTheScreen();
+      });
+
+      describe('press the backspace button', () => {
+        beforeEach(() => {
+          fireEvent.press(screen.getByTestId('back_space_button'));
+        });
+
+        it('should clear the display', () => {
+          const display = screen.getByTestId('formatted_number');
+          expect(display.props.children).toBe('');
+        });
+
+        it('should hide the backspace button', () => {
+          expect(
+            screen.queryByTestId('back_space_button'),
+          ).not.toBeOnTheScreen();
+        });
+
+        it('should disable the call button', () => {
+          const callButton = screen.getByTestId('call_button');
+          expect(callButton.props.accessibilityState.disabled).toBe(true);
+        });
+      });
+    });
+
+    describe('press more dialpad buttons', () => {
+      beforeEach(() => {
+        ['4', '1', '5', '6', '5', '0', '2', '8', '9', '0'].forEach((digit) => {
+          fireEvent.press(screen.getByText(digit));
+        });
+      });
+
+      it('should append the pressed dialpad digits to the display', () => {
+        expect(screen.getByTestId('formatted_number').props.children).toBe(
+          '+4156502890',
+        );
+      });
 
-  it('should allow user to use the backspace button', () => {
-    render(<Dialer />, { wrapper });
-    fireEvent.press(screen.getByText('4'));
-    fireEvent.press(screen.getByText('1'));
-    fireEvent.press(screen.getByText('5'));
-    expect(screen.getByText('+415')).toBeOnTheScreen();
-    fireEvent.press(screen.getByTestId('back_space_button'));
-    expect(screen.getByText('+41')).toBeOnTheScreen();
+      describe('press the backspace button', () => {
```

Comment:
do you still need this "backspace button" test with the "backspace button" tests on line 82-101?

Reply:
@kpchoy Those are two different scenarios:
1. Press backspace when there is only one digit on display
2. Press backspace when there are more than one digit on display